processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -55,6 +61,18 @@ public static void verifyAggregations(
             "Missing fields [%s] for postAggregator [%s]", missing, postAgg.getName()
         );
         Preconditions.checkArgument(combinedAggNames.add(postAgg.getName()), "[%s] already defined", postAgg.getName());
+
+        if (postAgg instanceof HasDependentAggFactories) {
```

Comment:
yeah, looking at this further, there seems to be other aggregators too which act as a wrapper for other postAggs e.g DoubleGreatestPostAggregator, LongGreatestPostAggregator etc. 

To make it more cleaner, I think it might just be simpler to add new method setDependentAggFactories in interface PostAggregator itself. Any implementation not having any dependent post aggregator will have a no-op implementation.